### PR TITLE
feat(SwissID): Add `Remove Google Play Integrity Integrity check` patch

### LINKED ISSUE #3477: feat (SwissId): Pass Integrity Check
### Feature description

package-name: com.swisssign.swissid.mobile

At the moment it is not possible to use this app with a custom rom installed, as the app required that you pass all [Integrity Checks](https://developer.android.com/google/play/integrity/overview)
It would be nice to have a patch that disabled this integrity check to allow the usage of SwissId on custom roms.

### Motivation

The app is currently unusable for many users, for example of custom ROMS like LineageOS or GrapheneOS
With a custom patch it should be possible to disable those limitations and allow the usage of the app for users of custom ROMS

### Acknowledgements

- [X] I have checked all open and closed feature requests and this is not a duplicate
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

### PR DESCRIPTION
**Pull request by KillTrot**
With this patch it is possible to use com.swisssign.swissid.mobile on custom roms as the integrity check will always return "VALID" for this app

Closes #3477

### CODE REVIEW

File: patches.json
Comment (oSumAtrIX) [inline patches.json:1]: Do not commit to this file

File: src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    compatiblePackages = [CompatiblePackage("com.swisssign.swissid.mobile", ["5.2.5"])],
+)
+@Suppress("unused")
+object RemoveIntegrityCheck : BytecodePatch(
+    setOf(IntegrityCheckFingerprint),
+) {
+    override fun execute(context: BytecodeContext) =
+        IntegrityCheckFingerprint.result?.mutableMethod?.addInstructions(
+            0,
+            "iget-object p1, p0, Lcom/swisssign/deviceintegrity/DeviceintegrityPlugin\$onMethodCall\$1;->\$result:Lio/flutter/plugin/common/MethodChannel\$Result;"
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:25]: Use multiline strings here

File: src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import app.revanced.patcher.patch.annotation.CompatiblePackage
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patches.swissid.integritycheck.fingerprints.IntegrityCheckFingerprint
+import app.revanced.util.exception
+
+@Patch(
+    name = "Remove device integrity check",
+    description = "Removes the Google Play Device Integrity Check." +
+        "With this it's possible to use SwissId on custom ROMS." +
+        "Rooted devices should hide the root from SwissId by putting SwissId on the DenyList",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:15]:
```suggestion
        "Rooted devices should hide the root from SwissId by putting SwissId on the DenyList.",
```

File: src/main/kotlin/app/revanced/patches/swissid/integritycheck/fingerprints/IntegrityCheckFingerprint.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import app.revanced.patcher.fingerprint.MethodFingerprint
+import com.android.tools.smali.dexlib2.AccessFlags
+import com.android.tools.smali.dexlib2.Opcode
+
+internal object IntegrityCheckFingerprint : MethodFingerprint(
+    returnType = "V",
+    parameters = listOf("Lcom/swisssign/deviceintegrity/model/DeviceIntegrityResult;"),
+    strings = listOf("it", "result"),
+    opcodes = listOf(
+        Opcode.CONST_STRING,
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/swissid/integritycheck/fingerprints/IntegrityCheckFingerprint.kt:12]: The pattern is too big. Reduce it to something where it won't easily break in future versions

File: src/main/kotlin/app/revanced/patches/swissid/integritycheck/fingerprints/IntegrityCheckFingerprint.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package app.revanced.patches.swissid.integritycheck.fingerprints
+
+import app.revanced.patcher.fingerprint.MethodFingerprint
+import com.android.tools.smali.dexlib2.AccessFlags
+import com.android.tools.smali.dexlib2.Opcode
+
+internal object IntegrityCheckFingerprint : MethodFingerprint(
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/swissid/integritycheck/fingerprints/IntegrityCheckFingerprint.kt:7]: Call this "CheckIntegrityFingerprint"

File: src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    compatiblePackages = [CompatiblePackage("com.swisssign.swissid.mobile", ["5.2.5"])],
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:16]: Is the version constraint necessary?

Comment (KillTrot) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:16]: I'm not sure, I would need to test it with older versions
Is a version constraint something generally not wanted if not strictly necessary?

File: src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt (hunk 1 above)
Comment (KillTrot) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:25]: I wanted to, but honestly I'm not able to make it work.
I must admit though, that Kotlin is not a language I regularly use.

If I do it like this:
```kotlin
override fun execute(context: BytecodeContext) =
        CheckIntegrityFingerprint.result?.mutableMethod?.addInstructions(
            0,
            """
            iget-object p1, p0, Lcom/swisssign/deviceintegrity/DeviceintegrityPlugin\$onMethodCall\$1;->\$result:Lio/flutter/plugin/common/MethodChannel\$Result;

            const-string v0, "VALID"

            invoke-interface {p1, v0}, Lio/flutter/plugin/common/MethodChannel\$Result;->success(Ljava/lang/Object;)V

            return-void
            """,
        ) ?: throw CheckIntegrityFingerprint.exception
```

It throws this exception when running `.\gradlew.bat generatePatchesFiles`:

```
KillTrot/revanced-patches/src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:26:87 Unresolved reference 'onMethodCall'.

KillTrot/revanced-patches/src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:26:107 Unresolved reference 'result'.
```

Maybe you can lead me the way on how to do it?

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:25]: The $ symbol needs to be escaped: `${"$"}`

File: src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt (hunk 5 above)
Comment (KillTrot) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:16]: Is it possible to set a min-version instead?

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:16]: Version names are arbitrary strings, no

Comment (KillTrot) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:16]: I understand.
I tried many versions now, and the first one that is working is version "5.0.1"
Older than that it does not work.

What do you think? Should I remove the version constraint or keep it?

Comment (KillTrot) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt:16]: That's not true, the last update was Jul. 12 2024

Comment (KillTrot) [reply src/main/kotlin/app/revanced/patches/swissid/integritycheck/RemoveIntegrityCheck.kt]: I removed the version constraint.
I don't know how you do it here, but if in a future version someone should have problems with the app, feel free to ping me :)

**Review by oSumAtrIX — APPROVED** [review]
pushed some refactoring changes. Confirm if everything works as expected. Lgtm apart from that.

**Comment by KillTrot** [comment]
The patching seems to work with the refactoring.

Strangely today I am not able to extract a usable apk from the xapk from apkpure anymore, but thats unrelated to this issue
But if someone has a way to extract or merge the xapk to an apk that doesn't throw this

```
com.android.apksig.apk.MinSdkVersionException: Failed to determine APK's minimum supported Android platform version
        at com.android.apksig.ApkSigner.getMinSdkVersionFromApk(ApkSigner.java:1006)
        at com.android.apksig.ApkSigner.sign(ApkSigner.java:295)
        at com.android.apksig.ApkSigner.sign(ApkSigner.java:231)
        at app.revanced.library.ApkSigner$Signer.signApk(ApkSigner.kt:337)
        at app.revanced.library.ApkUtils.signApk(ApkUtils.kt:169)
        at app.revanced.cli.command.PatchCommand.run(PatchCommand.kt:321)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at app.revanced.cli.command.MainCommandKt.main(MainCommand.kt:12)
Caused by: com.android.apksig.zip.ZipFormatException: Data Descriptor presence mismatch between Local File Header and Central Directory for entry AndroidManifest.xml. LFH: true, CD: false
        at com.android.apksig.internal.zip.LocalFileRecord.getRecord(LocalFileRecord.java:180)
        at com.android.apksig.internal.zip.LocalFileRecord.outputUncompressedData(LocalFileRecord.java:427)
        at com.android.apksig.internal.zip.LocalFileRecord.getUncompressedData(LocalFileRecord.java:451)
        at com.android.apksig.ApkSigner.getAndroidManifestFromApk(ApkSigner.java:966)
        at com.android.apksig.ApkSigner.getMinSdkVersionFromApk(ApkSigner.java:1004)
        ... 14 more
```

When signing, hit me up

**Pull request merged**